Shakefiles/Extra.hs

```diff
@@ -15,3 +16,12 @@ phonyPrefix prefix action =
 forEach :: Monad m => [a] -> (a -> m ()) -> m ()
 forEach list f =
     mapM_ f list
+
+
+--ghc-option required because of https://gitlab.haskell.org/ghc/ghc/-/issues/20592
+addMacOSGhcOptionFFI :: [String] -> [String]
+addMacOSGhcOptionFFI options =
+    case platform of
+        Mac -> ["--ghc-option=-I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/ffi"] ++ options
+        MacArm64 -> ["--ghc-option=-I/Library/Developer/CommandLineTools/SDKs/MacOSX11.sdk/usr/include/ffi"] ++ options
```

Comment:
The way to do that would be something like:

```haskell
StdoutTrim ffiFlags <- cmd "pkg-config" "--cflags" "libffi"
let ghcFfiFlags = "--ghc-option=" <> ffiFlags
```

(I think it's safe enough for now to assume the output won't contain stuff that needs to be escaped?  I'm not quite sure what'd be the easiest way to properly escape it.)

Reply:
Actually, `Development.Shake.cmd` will escape the arguments as needed, so maybe it will be escaped properly?  (Depends on how `--ghc-option` parses any stuff after the `=` 🤷 )